gqcp/include/QCMethod/CC/CCSDEnvironment.hpp

```diff
@@ -108,6 +122,27 @@ class CCSDEnvironment {
 
         return CCSDEnvironment<Scalar>(t1_amplitudes, t2_amplitudes, f, V_A);
     }
+
+    /**
+     *  Initialize a CCD algorithmic environment with initial guesses for the T2-amplitudes based on perturbation theory.
+     * 
+     *  @param sq_hamiltonian               the Hamiltonian expressed in an orthonormal spinor basis
+     *  @param orbital_space                the orbital space which covers the occupied-virtual separation
+     * 
+     *  @return an algorithmic environment suitable for coupled-cluster calculations up to the CCD level.
+     */
+    static CCSDEnvironment<Scalar> PerturbativeCCD(const SQHamiltonian<Scalar>& sq_hamiltonian, const OrbitalSpace& orbital_space) {
```

Comment:
Since you had to implement this named constructor, there seems to be a design error, since `PerturbativeCCD` and `Perturbative` exist both at the same time. There are three ways (I can think of right now) we can resolve this:
- Create a `CCDEnvironment`, analogous to the current `CCSDEnvironment`, but specifically tailored for CCD-calculations; or
- Create a `CCEnvironment` kind of factory function, with the APIs `PerturbativeCCSD` and `PerturbativeCCD`; or, the easiest option:
- Rename `CCSDEnvironment::Perturbative` to `CCSDEnvironment::PerturbativeCCSD`.

Reply:
Allright, I'm going for the 3rd option then. Wasn't sure if I could change the CCSD code.